[ML] Fix Single Metric Viewer forecast bounds being cut off in time slider in context chart

qn895 opened the pull request:
## Summary

This PR closes https://github.com/elastic/kibana/issues/91508. Changes include:
- Update the logic for the y scale to take into account the upper bound and lower bound of the model plot/forecast data
	
	Before
	<img width="648" alt="Screen Shot 2021-04-20 at 15 57 26" src="https://user-images.githubusercontent.com/43350163/115463195-25e6b900-a1f1-11eb-95cd-2a5eb8f94a5b.png">

	After
	<img width="648" alt="Screen Shot 2021-04-20 at 15 59 26" src="https://user-images.githubusercontent.com/43350163/115463419-67776400-a1f1-11eb-9337-2ca598b48a93.png">

- Fix the context chart not hiding the forecast data when the `show forecast` is toggled off
	
	Before
	<img width="648" alt="Screen Shot 2021-04-20 at 15 57 33" src="https://user-images.githubusercontent.com/43350163/115463265-3bf47980-a1f1-11eb-9b12-cad35082020d.png">

	After
	![2021-04-20 at 15 44](https://user-images.githubusercontent.com/43350163/115462958-e0c28700-a1f0-11eb-9084-68df0af8ffbc.gif)

qn895 committed:
[ML] Fix context chart not accounting for forecast data upper/lower bound as well as showForecast toggle

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121223/)
* Commit: bc7b532f6f65841b74d925b343e21536a229c9ed
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-97745/bc7b532f6f65841b74d925b343e21536a229c9ed)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121223/execution/node/719/log)
* [Execute kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121223/execution/node/580/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121223/","number":"121223","commit":"bc7b532f6f65841b74d925b343e21536a229c9ed"}]}
PIPELINE-->

qn895 commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'master' into ml-fix-context-chart-range-and-forecast

nreese committed:
[Maps] convert LayerControl to TS (#97728)

* [Maps] convert LayerControl to TS

* layer_control

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121420/)
* Commit: 4425406845449c204c272dacc1ab92205a581908
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [API Integration Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121420/execution/node/413/log)

### History
* :broken_heart: [Build #121223](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121223/) failed bc7b532f6f65841b74d925b343e21536a229c9ed

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121223/","number":"121223","commit":"bc7b532f6f65841b74d925b343e21536a229c9ed"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121420/","number":"121420","commit":"4425406845449c204c272dacc1ab92205a581908"}]}
PIPELINE-->

qn895 committed:
[ML] Fix Anomaly Detection job model memory limit input editable when datafeed is open (#97723)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

wayneseymour committed:
[QA] Switch tests to use importExport - discover (#94006)

Switch es archiver for kbn archiver everywhere possible

Partially closes #93797

spalger committed:
skip flaky suite (#97584)

stephmilovic committed:
[Security Solution] [Cases] Remove dynamic mappings for 3rd party cases mappings (#97754)

Commit:
Improve focus UX of MultiInputRows (#97695)

gchaps committed:
[DOCS] Reorganizes concepts and discover sections (#97019)

* [DOCS] Reorganizes concepts and discover sections

* [DOCS] Updates time filter and saved query sections

* Update docs/discover/save-search.asciidoc

Co-authored-by: Wylie Conlon <wylieconlon@gmail.com>

* [DOCS] Updates time filter and saved query doc

* [DOCS] Updates images

* [DOCS] Adds missing images

* [DOCS] Minor edits

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Wylie Conlon <wylieconlon@gmail.com>

mistic committed:
chore(NA): adds 7.13 branch and bumps 7.x on backportrc (#97804)

darnautov committed:
[ML] UI enhancements for Anomaly detection rule type  (#97626)

* [ML] update labels

* [ML] update job summary endpoint to return associated alert rules

* [ML] add alert rule icon to the table

* [ML] edit alert rules from ML UI

* [ML] register navigation

* [ML] support single job selection only

* [ML] remove groups options from the job selection

* [ML] deps on rule id to avoid re-rendering

* [ML] fix i18n

* [ML] add info message to the alert context

* [ML] fix typo

* [ML] register usage collection

* [ML] fix telemetry

spalger committed:
skip flaky suite (#97864)

Commit:
Remove the no longer used release-notes script (#97806)

* Remove the no longer used release-notes script

* Commit missing file

Commit:
[npm] upgrade all jest related dependencies (#96367)

Co-authored-by: spalger <spalger@users.noreply.github.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
chore(NA): chore(NA): moving @kbn/std into bazel (#97771)

Commit:
Add the more explicit & actionable text listed in ticket (#97857)

## Summary
closes #92591

Create a new `HostedAgentPolicyRestrictionRelatedError` which ensures the additional text from #92591 is included in all the appropriate places, but only specified once.

#### Some current examples
 * `Cannot update integrations of hosted agent policy ${id}`
 * `Cannot remove integrations of hosted agent policy ${id}`
 * `Cannot reassign an agent to hosted agent policy ${newAgentPolicy.id}`


#### In this PR are now
 * `Cannot update integrations of hosted agent policy ${id}  in Fleet because the agent policy is managed by an external orchestration solution, such as Elastic Cloud, Kubernetes, etc. Please make changes using your orchestration solution.`
 * `Cannot remove integrations of hosted agent policy ${id} in Fleet because the agent policy is managed by an external orchestration solution, such as Elastic Cloud, Kubernetes, etc. Please make changes using your orchestration solution.`
 * `Cannot reassign an agent to hosted agent policy ${newAgentPolicy.id} in Fleet because the agent policy is managed by an external orchestration solution, such as Elastic Cloud, Kubernetes, etc. Please make changes using your orchestration solution.`


### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

spalger committed:
remove jest-environment-jsdom-thirteen from package.json

marshallmain committed:
[Security Solution][Detections] Fix flaky threshold API tests (#97768)

* Explicitly refreshes signals index for threshold api tests to prevent flakiness

* Unskip test suite

cauemarcondes committed:
[APM] add comparison to Instances latency distribution (#97710)

* adding comparion data to chart

* fixing api test

* addressing comments

* refactoring

cauemarcondes committed:
[APM] Adding comparison data to the dependencies table (#97843)

* adding comparison to dependencies api

* fixing client

* fixing client

* changing client

nchaulet committed:
[Fleet] Always create agent policy as active (#97874)

alvarezmelissa87 committed:
[ML][Transform] display script_exception compile error message in wizards (#97697)

* ensure error rootCause script gets added to message

* ensure script part of error is shown in DFA wizard

* use isPopulatedObject to check for script field

tylersmalley committed:
[kbn-es] Increase ES heap (#97905)

Signed-off-by: Tyler Smalley <tyler.smalley@elastic.co>

spalger committed:
bump 7.x version in renovate config

jbudz committed:
[cli/keystore] Transpile sources in dev (#97501)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

spalger committed:
bump version label in elastic/charts section of renovate config

jbudz committed:
[deb/rpm] Cleanup log and pid folders on package uninstall (#97884)

This is a consistency check with elasticsearch, which removes the log
and pid folders on package uninstall.

Part of #86042

JasonStoltz committed:
[App Search] Remove unknowns: 'allow' from enterprise_search routes (#97510)

* Remove unknown allows from enterprisesearch routes

It is a best practice to avoid `unknowns: allow` in request
body validations.

Usage of `unknowns: allow` in the App Search plugin has primarily
been to avoid creating full validation schemas, as they can often
be complex.

Rather than have a half-baked validation that uses `unknowns: allow`,
we have the option to skip parsing and validating a JSON body entirely,
and simply pass it through "as-is" to the enterprise_search server
for validation.

* Move to helper

* Swap comment block style

* PR feedback

* Update x-pack/plugins/enterprise_search/server/lib/route_config_helpers.test.ts

Co-authored-by: Constance <constancecchen@users.noreply.github.com>

* Update x-pack/plugins/enterprise_search/server/lib/route_config_helpers.test.ts

Co-authored-by: Constance <constancecchen@users.noreply.github.com>

* Update x-pack/plugins/enterprise_search/server/lib/route_config_helpers.ts

Co-authored-by: Constance <constancecchen@users.noreply.github.com>

* Update x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.ts

Co-authored-by: Constance <constancecchen@users.noreply.github.com>

* Better TS

* Change to TS validations

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Constance <constancecchen@users.noreply.github.com>

mattkime committed:
set initial STRING for confirm box (#97907)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

angorayc committed:
fix tooltip content (#97847)

dasansol92 committed:
[Security Solution] User can select event from event list and create a filter (#96940)

* Initial version of event filtering form/dialog. Pending to add all redux services

* Uses redux store instead of props to get the form values

* Manage errors on redux

* Creates even filter list on service constructor

* Add os type selector depending on form parent by props. Also added create action

* Allows add exception to an event. This commit has to be reviewed and maybe it will change depending on next changes

* Fix imports because changes on ExceptionBuilder component and add needed type export

* Adds constants. Rename eventFilters to eventFilter. Add http wrapper as a hook to check if the list has been created or not

* Adds missing files on last commit.

* Relocate async resource state to be shared between different pages

* Use async resource state to manage async operations on components. Relocate initial entry status to an utils module instead of hook.

* Adds comments into redux store from component

* Fixes typechecks and wrong imports

* Fixes translations and adds subheader and description modal

* Relocates form description

* Removes unused import

* Sanitize entries before submit to remove entry.id

* Missed file on last commit

* Use specific fields for endpoint_event type builder

* Split error field for each kind of errors to prevent unexpected renders. Adds unit test for event filter form component

* Set event.kind == event by default

* Changes folder names. Add notifications when success. Remove default event.king

* Adds notifications when api error and fixed multiple notifications showed for same error

* Adds new test for event filter modal and changes component name to be consistent

* Adds unit tests for event filter notification

* Adds middleware unit tests. Also isolate common event for all tests

* Adds unit tests for event filter reducer

* Adds unit tests for event filter selector

* Fixes same key on different multilanguages. Fixes naming incoherence

* Adds feature flag for event filtering

* Fixes unit tests and weird behavior when changing items after name or comments on event filter form

* Removes unused import

* Fixes unit tests. Add imports from lists plugin. Add expects on tests. Change some names

* Renames everything from eventFilter to eventFilters (plural)

* Rename state variable

* Create hook for notifications instead of a component. Removes className from modal body.

* Updates available fields for enpoint events builder

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

peteharverson committed:
[ML] Indicate Apache and Nginx modules as legacy versions (#97801)

* [ML] Indicate Apache and Nginx modules as legacy versions

* [ML] Edits to module titles to capitalize Filebeat

Commit:
Fix broken advanced setting i18n (#97782)

peteharverson commented on x-pack/plugins/ml/public/application/timeseriesexplorer/components/timeseries_chart/timeseries_chart.js:
```diff
@@ -982,11 +984,14 @@ class TimeseriesChartIntl extends Component {
     each(combinedData, (item) => {
-      valuesRange.min = Math.min(item.value, valuesRange.min);
-      valuesRange.max = Math.max(item.value, valuesRange.max);
+      const lowerBound = item.lower ?? Number.MIN_VALUE;
```
I think these are the wrong way round, and should be 

```
      const lowerBound = item.lower ?? Number.MAX_VALUE;
      const upperBound = item.upper ?? Number.MIN_VALUE;
```

Otherwise for the case where there is no model plot you see this with your change:
![image](https://user-images.githubusercontent.com/7405507/115691919-657ae580-a356-11eb-9c3e-440004f7cefa.png)

Before your edit:
<img width="1551" alt="smv_context_1" src="https://user-images.githubusercontent.com/7405507/115691711-35334700-a356-11eb-8c39-13b602cd907e.PNG">

lizozom committed:
[Search] Remove msearch from searchsource (#96936)

* Move inspector adapter integration into search source

* docs and ts

* Move other bucket to search source

* test ts + delete unused tabilfy function

* hierarchical param in aggconfig.
ts improvements
more inspector tests

* fix jest

* separate inspect
more tests

* jest

* inspector

* Error handling and more tests

* put the fun in functional tests

* delete client side legacy msearch code

* ts

* override to sync search in search source

* delete more legacy code

* ts

* delete moarrrr

jloleysens committed:
[Ingest Pipelines] Fix descriptions not showing and minor appearance tweaks (#97799)

* fix descriptions not showing and minor appearance tweaks

* remove unnecessary prop

* added jest test for presence of descriptions

* refactored variable names

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Kerry350 committed:
Use create over update (#97818)

angorayc committed:
[Security Solution] Events integration test (#97519)

* events integration test

* add constants

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

afharo committed:
[flaky-test] Unskip SOM edit_saved_object tests (#97846)

scottybollinger committed:
[Workplace Search] PR#3358 to Kibana (#97921)

* Render flash message when server sends back an error

* Fetch data on all route changes

This produced a bug where the nav and loading states were triggered between route changes. Added conditional to prevent resetting between in-source changes

pgayvallet committed:
Block value `0` for terms.min_doc_count aggregation (#97966)

* Block value `0` for terms.min_doc_count aggregation

* do the same for histogram

nreese committed:
[Maps] fix cannot read propery 'getImage' of undefined (#97829)

qn895 committed:
[ML] Fix Data Visualizer event rate chart empty for some indices when using long time range (#97655)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

qn895 committed:
[ML] Fix the bounds not correct

qn895 committed:
Merge branch 'ml-fix-context-chart-range-and-forecast' of https://github.com/qn895/kibana into ml-fix-context-chart-range-and-forecast

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)
